src/pde.cpp

```diff
@@ -151,8 +177,56 @@ void PDE::set_solution (
     _solution_function->interpolate(*constant_fn);
   }
 
-  _bilinear_form->set_coefficient("uu", _solution_function);
-  _linear_form->set_coefficient("uu", _solution_function);
+  _bilinear_form->set_coefficient(_variable, _solution_function);
+  _linear_form->set_coefficient(_variable, _solution_function);
+}
+//--------------------------------------
+void PDE::set_solutions (
+  std::vector<Linear_Function> expression
+) {
+
+  std::size_t dimension = expression.size();
+  //
+  // printf("\t\t SIZE = %d \n",expression.size());
+  // printf("\t\t SIZE = %d \n",_variables.size());
+  // printf("\t\t SIZE = %d \n",_solution_functions.size());
+
+  if ( (_variables.size() == dimension) && (_solution_functions.size() == dimension) ){
+
+    for (std::size_t i = 0; i < dimension; i++) {
+      _solution_functions[i].reset(new dolfin::Function(_functions_space[i]));
+      _solution_functions[i]->interpolate(expression[i]);
+      _linear_form->set_coefficient(_variables[i], _solution_functions[i]);
+    }
+    _bilinear_form->set_coefficient(_variables[0], _solution_functions[0]);
+    _bilinear_form->set_coefficient(_variables[1], _solution_functions[1]);
+
+  }
+  else {
+    printf("Dimension mismatch!!\n");
+  }
+}
+//--------------------------------------
+void PDE::set_solutions (
+  std::vector<dolfin::Function> new_solutions
+) {
+
+  std::size_t dimension = new_solutions.size();
+
+  if (_solution_functions.size() == dimension) {
+
+    for (std::size_t i = 0; i < dimension; i++) {
+      _solution_function.reset(new dolfin::Function(_functions_space[i]));
+      *(_solution_functions[i])= new_solutions[i];
+      // _bilinear_form->set_coefficient(_variables[i], _solution_functions[i]);
+      _linear_form->set_coefficient(_variables[i], _solution_functions[i]);
```

Comment:
That's for Coefficient and Source which is fine. But the PNP+Stokes cases, the variables are separated and the pressure does not appear in the bilinear form. To make it more general we might want to have two variables mapping one for the bilinear form, one for the linear form like Coeff and Source.